packages/rocketchat-lib/client/lib/roomTypes.js

```diff
@@ -59,7 +59,20 @@ RocketChat.roomTypes = new class RocketChatRoomTypes extends RoomTypesCommon {
 				_id: 1,
 			},
 		});
-		return room && (room.ro === true && Array.isArray(room.muted) && room.muted.indexOf(user.username) !== -1 && !userOwner);
+
+		if (userOwner) {
+			return false;
+		}
+
+		if (room) {
+			if (Array.isArray(room.muted) && room.muted.indexOf(user.username) !== -1) {
+				return true;
+			}
+
+			return (room.ro === true && !RocketChat.authz.hasAtLeastOnePermission('post-readonly'));
```

Comment:
`hasAtLeastOnePermission` is missing the `scope` param, because of this _moderators_ cannot talk on a read-only room even with the permission